### DISCUSSION TITLE
index

### DISCUSSION THREAD

**Discussion by imroc** [imroc/kubernetes-guide#84] [General]
https://imroc.cc/kubernetes

**Comment by imroc** [comment]
干货装载中，尽情期待~

**Comment by sunface** [comment]
快快装完，准备上车

**Comment by imroc** [comment]
> @sunface
> 快快装完，准备上车

快了快了 😎

**Comment by sunface** [comment]
> > @sunface
> > 快快装完，准备上车
> 
> 快了快了 😎

未来的 k8s 巨著，反正我是第一个 star 的，哈哈

**Comment by imroc** [comment]
> @sunface
> 未来的 k8s 巨著，反正我是第一个 star 的，哈哈

好眼光 👍 电子书还是参考你的 [Rust语言圣经](https://course.rs/) 搞的，自己再搞了评论和文章内导航的功能。

**Comment by imroc** [comment]
> 好眼光 👍 电子书还是参考你的 [Rust语言圣经](https://course.rs/) 搞的，自己再搞了评论和文章内导航的功能。

@sunface 哈哈，注意到你已经把这俩功能借鉴过去了 🤓 不过你的书没有固定子路径，评论的 js 需要微调一点点。

**Comment by sunface** [comment]
好，我是见猎心喜，就赶紧抄过来了 😋

**Comment by sunface** [comment]
> > @sunface
> > 未来的 k8s 巨著，反正我是第一个 star 的，哈哈
> 
> 好眼光 👍 电子书还是参考你的 [Rust语言圣经](https://course.rs/) 搞的，自己再搞了评论和文章内导航的功能。

不过我发现这个章节内 toc 只有在宽屏幕上才有，14寸的笔记本电脑上就没有，怎么破？

**Comment by imroc** [comment]
> @sunface
> 不过我发现这个章节内 toc 只有在宽屏幕上才有，14寸的笔记本电脑上就没有，怎么破？

不会吧，我 Mac M1 13.3 寸都可以显示，正常分辨率。你可以调 `style.css`，当前是:

```css
@media only screen and (min-width:1440px)
```

**Comment by sunface** [comment]
知道原因了，我在 macos 设置里，使用了缩放，这样字更大一些 ：）

**Comment by imroc** [comment]
> @sunface
> 知道原因了，我在 macos 设置里，使用了缩放，这样字更大一些 ：）

👍

**Comment by sunface** [comment]
```css
@media only screen and (min-width:1080px) {
    main {
        position: relative;
        padding-right: 170px;
    }
```

我修改了最小屏幕的尺寸，然后修复了自适应的问题：当屏幕较小时，toc 会 float 遮住章节的主体文字，你可以参考下～～

**Comment by imroc** [comment]
> @sunface
> ```css
> @media only screen and (min-width:1080px) {
>     main {
>         position: relative;
>         padding-right: 170px;
>     }
> ```
> 
> 我修改了最小屏幕的尺寸，然后修复了自适应的问题：当屏幕较小时，toc 会 float 遮住章节的主体文字，你可以参考下～～

👍 nice，很好用，已经抄过来了~

**Comment by imroc** [comment]
@sunface  哈哈，评论系统我又无缝迁移到 [giscus](https://giscus.app/zh-CN) 了，功能更强大。利用 github discussion API，不需要提前创建 issue 或 discussion，有人评论时才自动生成，回复也不需要引用之前内容。欢迎来抄作业~

**Reply by sunface** [reply]
在你评论前就已经发现了。。明天开抄 = ， = 哈哈

**Reply by sunface** [reply]
确实这个要好很多～～以后我就盯准你的网站了，看看有什么黑科技哈哈。

**Reply by sunface** [reply]
对了，现有的 page-toc css 有 bug，在左侧目录开启的情况下，右侧的章节内 toc 中的目录项点击不到。

原因是 page-toc 被 main 元素遮挡了，你需要设置下 z-index

**Reply by imroc** [reply]
敏锐的洞察力 😏

**Reply by imroc** [reply]
> 原因是 page-toc 被 main 元素遮挡了，你需要设置下 z-index

你不说我还没注意到有这个bug，大概是我平时用的屏幕尺寸太大了，测试了下缩短浏览器宽度后复现了，然后抄了你的 `z-index` 修复了，开源果然是个好东西 😎

**Reply by sunface** [reply]
giscus的输入框里的光标左右移动事件可以被优先捕获，之前那个，左右移动直接跳转页面了 = , =

**Reply by imroc** [reply]
确实是~

**Reply by sunface** [reply]
我在你的基础上稍微优化了下：

- 获取浏览器的当前语言作为 giscuz 的语言
- 根据当前 mdbook 的主题设置 giscuz 主题，但是切换 mdbook 主题后，要刷新页面重新加载 js 后才能生效

**Reply by imroc** [reply]
太棒了，不过我觉得书籍本身定位是中文，就没必要改变 giscus 语言。
我借鉴了你的思路，进一步做了优化：giscus 主题跟随 mdbook 主题，可以自动变化，不需要重新加载。

参考这个 [commit](https://github.com/imroc/kubernetes-guide/commit/9f4da9a0e2ac9e70586085142fc8e0f01ae0faa8)

**Reply by sunface** [reply]
666，给 js 大佬跪了

**Comment by scientiacoder** [comment]
看起来不错

**Comment by cuisongliu** [comment]
# 下载并安装sealos, sealos是个golang的二进制工具，直接下载拷贝到bin目录即可, release页面也可下载
wget -c https://sealyun-home.oss-cn-beijing.aliyuncs.com/sealos-4.0/latest/sealos-amd64 -O sealos && \
    chmod +x sealos && mv sealos /usr/bin
# 下载buildah
wget https://sealyun-home.oss-accelerate.aliyuncs.com/images/buildah.linux.amd64 --no-check-certificate -O buildah && chmod a+x buildah && mv buildah /usr/bin
# 创建一个集群
sealos run registry.cn-hongkong.aliyuncs.com/sealyun/oci-kubernetes:1.22.9 registry.cn-hongkong.aliyuncs.com/sealyun/oci-calico:v3.22.1 --masters 192.168.64.2,192.168.64.22,192.168.64.20 --nodes 192.168.64.21,192.168.64.19

一键安装不香么，哈哈哈

**Comment by laiqinghua** [comment]
大佬强强强！带我们冲冲冲！

**Comment by sologgfun** [comment]
干货很多，不过网络排障没有看到calico相关，建议添加

**Reply by imroc** [reply]
慢慢来，都是挤空余时间搞

**Comment by waidinsamkeit** [comment]
### 大佬！！！

**Comment by waidinsamkeit** [comment]
大佬

**Comment by rockburning** [comment]
大佬 这个可以转成pdf吗

**Reply by imroc** [reply]
不行，站点基于 docusaurus，没有好用的转 pdf 插件，而且包含一些自定义组件，就更无法实现转pdf

**Reply by rockburning** [reply]
> 不行，站点基于 docusaurus，没有好用的转 pdf 插件，而且包含一些自定义组件，就更无法实现转pdf

哦哦 好的 谢谢 想着可以转成pdf的话就很nice
